lib/puppet/file_bucket/dipper.rb

```diff
@@ -56,6 +58,38 @@ def backup(file)
     end
   end
 
+  # Diffs two filebucket files identified by their sums
+  def diff(suma, sumb)
+    raise RuntimeError, "Diff is not supported on windows platforms" if Puppet.features.microsoft_windows?
+    source_path = "#{@rest_path}#{@checksum_type}/#{suma}"
+    if sumb and sumb != ''
+      file_diff = Puppet::FileBucket::File.indirection.find(
+        source_path,
+        :bucket_path => @local_path,
+        :diff_with => sumb)
+    else
+      original_paths = Puppet::FileBucket::File.indirection.find(
+        source_path,
+        :bucket_path => @local_path,
+        :original_path => true)
+      raise Puppet::Error, "Couldn't retrieve file's original path" unless original_paths
+      original_paths = original_paths.to_s.split(":")
+      file_diff = ""
+      original_paths.each do |original_path|
```

Comment:
So you're suggesting to add an another optionnal parameter `--diff-with` (or whatever parameter pointing to a file) which will be diffed with the file in the filebucket. And, by default, it will try to find the "original path" using the `paths` file and if he finds multiple paths it will just fail. Is that right ?